test/test-fs-event.c

```diff
@@ -489,26 +490,36 @@ TEST_IMPL(fs_event_watch_dir_short_path) {
   create_dir("watch_dir");
   create_file("watch_dir/file1");
 
-  r = uv_fs_event_init(loop, &fs_event);
-  ASSERT(r == 0);
-  r = uv_fs_event_start(&fs_event, fs_event_cb_dir, "watch_~1", 0);
-  ASSERT(r == 0);
-  r = uv_timer_init(loop, &timer);
-  ASSERT(r == 0);
-  r = uv_timer_start(&timer, timer_cb_file, 100, 0);
-  ASSERT(r == 0);
-
-  uv_run(loop, UV_RUN_DEFAULT);
+  /* Newer version of Windows ship with HKLM\SYSTEM\CurrentControlSet\Control\FileSystem\NtfsDisable8dot3NameCreation != 0
+   * So we verify the files we created are addressable by a 8.3 short name */
+  r = uv_fs_stat(NULL, &req, "watch_~1", NULL);
+  if (r != UV_ENOENT) {
+    r = uv_fs_event_init(loop, &fs_event);
+    ASSERT(r == 0);
+    r = uv_fs_event_start(&fs_event, fs_event_cb_dir, "watch_~1", 0);
+    ASSERT(r == 0);
+    r = uv_timer_init(loop, &timer);
+    ASSERT(r == 0);
+    r = uv_timer_start(&timer, timer_cb_file, 100, 0);
+    ASSERT(r == 0);
+  
+    uv_run(loop, UV_RUN_DEFAULT);
 
-  ASSERT(fs_event_cb_called == 1);
-  ASSERT(timer_cb_called == 1);
-  ASSERT(close_cb_called == 1);
+    ASSERT(fs_event_cb_called == 1);
+    ASSERT(timer_cb_called == 1);
+    ASSERT(close_cb_called == 1);
+  }
 
   /* Cleanup */
   remove("watch_dir/file1");
   remove("watch_dir/");
 
   MAKE_VALGRIND_HAPPY();
+  
+  /* `UV_ENOENT` could only be from when we tried to open with an 8.3 name. */
+  if (r == UV_ENOENT) {
```

Comment:
Would you be open to using a separate variable here? I just worry about code being added to the test in the future and overwriting `r`.

Reply:
Ack.
Introduced `has_shortnames`, which is self documenting, so removed comment.